### PR TITLE
Units and Quantities lesson

### PR DESCRIPTION
**Pull request by dpshelio**
This is the first almost complete draft, I may think on adding more exercises at the end, but I don't think in 45 minutes much more can be taught...

Please, review :)

### CODE REVIEW

File: xx-units/Units_Quantities.ipynb [hunk 1]
```diff
@@ -0,0 +1,668 @@
...
+   "metadata": {
+    "collapsed": false,
+    "keep": true
+   }
+  },
+  {
+   "execution_count": 2,
+   "cell_type": "code",
+   "source": [
+    "# Deffine a quantity length\n",
```
Comment (SolarDrew) [inline xx-units/Units_Quantities.ipynb]: Deffine -> Define

File: xx-units/Units_Quantities_Instructor.ipynb [hunk 2]
```diff
@@ -0,0 +1,1198 @@
...
+     "text": [
+      "26.2 m\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Deffine a quantity length\n",
+    "length = 26.2 * u.meter\n",
+    "# print it\n",
+    "print(length) # lenght is a quantity"
```
Comment (SolarDrew) [inline xx-units/Units_Quantities_Instructor.ipynb]: lenght -> length

File: xx-units/Units_Quantities_Instructor.ipynb [hunk 3]
```diff
@@ -0,0 +1,1198 @@
...
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "25.5555444444 m / s\n",
+      "25.5555444444 m / s\n"
+     ]
+    }
+   ],
+   "source": [
+    "# decompos it\n",
```
Comment (SolarDrew) [inline xx-units/Units_Quantities_Instructor.ipynb]: decompos -> decompose

File: xx-units/Units_Quantities_Instructor.ipynb [hunk 4]
```diff
@@ -0,0 +1,1198 @@
...
+    " <li> Convert the speed in imperial units (miles/hour) using: <br>\n",
+    "\n",
+    " ```from astropy.units import imperial```\n",
+    " </li>\n",
+    " <li> Calculate whether a pint is more than half litter <br>\n",
+    " \n",
+    " <emph>You can compare quantities as comparing variables.</emph> <br>\n",
+    " Something strange? Check what deffinition of <a href='https://en.wikipedia.org/wiki/Pint'>pint</a> astropy is using.\n",
+    " </li>\n",
+    " <li> Does units work with areas? calculate the area of a rectangle of 3 km of side and 5 meter of width. Show them in m2 and convert them to yards2</li>\n",
```
Comment (SolarDrew) [inline xx-units/Units_Quantities_Instructor.ipynb]: m^2, yards^2

File: xx-units/Units_Quantities.ipynb [hunk 5]
```diff
@@ -0,0 +1,668 @@
...
+   "source": [
+    "<div style='background:#B1E0A8; padding:10px 10px 10px 10px;'>\n",
+    "<H2> Challenges </H2>\n",
+    "\n",
+    " <ol>\n",
+    " <li> Convert the speed in imperial units (miles/hour) using: <br>\n",
+    "\n",
+    " ```from astropy.units import imperial```\n",
+    " </li>\n",
+    " <li> Calculate whether a pint is more than half litter <br>\n",
```
Comment (SolarDrew) [inline xx-units/Units_Quantities.ipynb]: litter -> litre

File: xx-units/Units_Quantities.ipynb [hunk 6]
```diff
@@ -0,0 +1,668 @@
...
+    "Quantities can also be applied to arrays"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "execution_count": 29,
+   "cell_type": "code",
+   "source": [
+    "# different ways of deffining a quantity for a single value\n"
```
Comment (SolarDrew) [inline xx-units/Units_Quantities.ipynb]: deffining -> defining

File: xx-units/Units_Quantities.ipynb [hunk 7]
```diff
@@ -0,0 +1,668 @@
...
+   "outputs": [],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "source": [
+    "## Creating functions with quantities as units\n",
+    "\n",
+    "We want to have functions that contains the information of the untis, and with them we can be sure that we will be always have the *right* result."
```
Comment (SolarDrew) [inline xx-units/Units_Quantities.ipynb]: contains -> contain

File: xx-units/Units_Quantities_Instructor.ipynb [hunk 8]
```diff
@@ -0,0 +1,1198 @@
...
+      ]
+     },
+     "execution_count": 35,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# run with and without units\n",
+    "kinetic(5, 10) # Fails! it doesn't specifies the units.\n",
```
Comment (SolarDrew) [inline xx-units/Units_Quantities_Instructor.ipynb]: specifies -> specify

File: xx-units/Units_Quantities_Instructor.ipynb [hunk 9]
```diff
@@ -0,0 +1,1198 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div style='background:#B1E0A8; padding:10px 10px 10px 10px;'>\n",
+    "<H2> Challenges </H2>\n",
+    "\n",
+    " <ol start=4>\n",
+    "  <li> Create a function that calculates potential energy where *g* defaults for Earth value, \n",
```
Comment (SolarDrew) [inline xx-units/Units_Quantities_Instructor.ipynb]: defaults for -> defaults to

File: xx-units/Units_Quantities_Instructor.ipynb [hunk 10]
```diff
@@ -0,0 +1,1198 @@
...
+       "<Quantity 9.375 J>"
+      ]
+     },
+     "execution_count": 38,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# in mars:\n",
```
Comment (SolarDrew) [inline xx-units/Units_Quantities_Instructor.ipynb]: on Mars

File: xx-units/Units_Quantities_Instructor.ipynb [hunk 11]
```diff
@@ -0,0 +1,1198 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div style='background:#B1E0A8; padding:10px 10px 10px 10px;'>\n",
+    "<H2> Challenges </H2>\n",
+    "\n",
+    " <ol start=5>\n",
+    "  <li> Convert the area calculated before `rectangle_area` in <a href=\"https://en.wikipedia.org/wiki/Hectare\">Hectare</a>\n",
+    "  (1 hectare = 100 ares; 1 are = 100 m2).\n",
```
Comment (drewleonard42) [inline xx-units/Units_Quantities_Instructor.ipynb]: ares -> acres, presumably?
are -> acre

**Comment by SolarDrew** [comment]
Looks good to me. Apologies for my pedantic comments, @dpshelio :P

**Comment by dpshelio** [comment]
Thanks @drewleonard42!! all corrected now :)

**Pull request merged**